### PR TITLE
CXP-1421: Display credentials in the custom apps settings

### PR DESCRIPTION
**Pull request by eoliveauakeneo**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Marketplace/TestApps/Controller/Internal/GetCustomAppSecretAction.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+use Oro\Bundle\SecurityBundle\SecurityFacade;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/**
+ * @copyright 2022 Akeneo SAS (http://www.akeneo.com)
```
Comment (guillaumefer-akeneo) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Marketplace/TestApps/Controller/Internal/GetCustomAppSecretAction.php:18]:
```suggestion
 * @copyright 2023 Akeneo SAS (http://www.akeneo.com)
```
:smile:

File: src/Akeneo/Connectivity/Connection/back/tests/EndToEnd/Marketplace/TestApps/Controller/Internal/GetCustomAppSecretActionEndToEnd.php [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+use Akeneo\Connectivity\Connection\back\tests\EndToEnd\WebTestCase;
+use Akeneo\Connectivity\Connection\Infrastructure\Marketplace\TestApps\Persistence\GetTestAppSecretQuery;
+use Akeneo\Connectivity\Connection\Tests\CatalogBuilder\TestAppLoader;
+use Akeneo\Platform\Bundle\FeatureFlagBundle\Internal\Test\FilePersistedFeatureFlags;
+use Akeneo\Test\Integration\Configuration;
+use PHPUnit\Framework\Assert;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * @copyright 2022 Akeneo SAS (http://www.akeneo.com)
```
Comment (guillaumefer-akeneo) [inline src/Akeneo/Connectivity/Connection/back/tests/EndToEnd/Marketplace/TestApps/Controller/Internal/GetCustomAppSecretActionEndToEnd.php:16]:
```suggestion
 * @copyright 2023 Akeneo SAS (http://www.akeneo.com)
```

File: src/Akeneo/Connectivity/Connection/back/tests/EndToEnd/Marketplace/TestApps/Controller/Internal/GetCustomAppSecretActionEndToEnd.php [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    }
+
+    protected function getConfiguration(): Configuration
+    {
+        return $this->catalog->useMinimalCatalog();
+    }
+
+    public function test_it_gets_secret(): void
+    {
+        $this->featureFlags->enable('app_developer_mode');
```
Comment (guillaumefer-akeneo) [inline src/Akeneo/Connectivity/Connection/back/tests/EndToEnd/Marketplace/TestApps/Controller/Internal/GetCustomAppSecretActionEndToEnd.php:41]: No need developer mode anymore in internal controller ? Can be removed in the next PR dedicated to the developer mode removal.

**Review by guillaumefer-akeneo — APPROVED** [review]

File: src/Akeneo/Connectivity/Connection/front/src/connect/hooks/use-fetch-custom-app-secret.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+type Error = string | null;
+type Result = {
+    isLoading: boolean;
+    isError: boolean;
+    data: string | undefined;
+    error: Error;
+};
+
+export const useFetchCustomAppSecret = (customAppId: string): Result => {
+    return useQuery<string, Error, string>(['id', customAppId], async () => {
```
Comment (RiffVentura) [inline src/Akeneo/Connectivity/Connection/front/src/connect/hooks/use-fetch-custom-app-secret.ts:12]:
```suggestion
    return useQuery<string, Error, string>(['custom_app_secret', customAppId], async () => {
```

**Review by RiffVentura — APPROVED** [review]

File: src/Akeneo/Connectivity/Connection/front/src/connect/components/ConnectedApp/ConnectedAppCredentials.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+import {ConnectedApp} from '../../../model/Apps/connected-app';
+import {KeyIcon, SectionTitle, UserIcon} from 'akeneo-design-system';
+import {useTranslate} from '../../../shared/translate';
+import styled from 'styled-components';
+import {CopiableCredential} from '../../../settings/components/credentials/CopiableCredential';
+import {Credential} from '../../../settings/components/credentials/Credential';
+import {useFetchCustomAppSecret} from '../../hooks/use-fetch-custom-app-secret';
+
+export const CredentialList = styled.div`
+    display: grid;
```
[the remark this answers is not in the document]
Comment (eoliveauakeneo) [reply src/Akeneo/Connectivity/Connection/front/src/connect/components/ConnectedApp/ConnectedAppCredentials.tsx:11]: I reused the code from the connexions CredentialList

**Pull request merged**